src/distribution/logistic.rs

```diff
@@ -0,0 +1,454 @@
+        f64::NEG_INFINITY
+    }
+}
+
+impl Max<f64> for Logistic {
+    /// Returns the maximum value in the domain of the logistic
+    /// distribution representable by a double precision float
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// INF
+    /// ```
+    fn max(&self) -> f64 {
+        f64::INFINITY
+    }
+}
+
+impl Distribution<f64> for Logistic {
+    /// Returns the mode of the logistic distribution
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// μ
+    /// ```
+    ///
+    /// where `μ` is the location
+    fn mean(&self) -> Option<f64> {
+        Some(self.location)
+    }
+    /// Returns the variance of the logistic distribution
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// σ * π^2 / 3
+    /// ```
+    ///
+    /// where `σ` is the scale
+    fn variance(&self) -> Option<f64> {
+        Some(self.scale.powi(2) * consts::PI_SQ / 3.0)
+    }
+    /// Returns the entropy of the logistic distribution
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// ln(σ) + 2
+    /// ```
+    ///
+    /// where `σ` is the scale
+    fn entropy(&self) -> Option<f64> {
+        Some(self.scale.ln() + 2.0)
+    }
+    /// Returns the skewness of the logistic distribution
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// 0
+    /// ```
+    fn skewness(&self) -> Option<f64> {
+        Some(0.0)
+    }
+}
+
+impl Median<f64> for Logistic {
+    /// Returns the median of the logistic distribution
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// μ
+    /// ```
+    ///
+    /// where `μ` is the location
+    fn median(&self) -> f64 {
+        self.location
+    }
+}
+
+impl Mode<Option<f64>> for Logistic {
+    /// Returns the mode of the logistic distribution
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// μ
+    /// ```
+    ///
+    /// where `μ` is the location
+    fn mode(&self) -> Option<f64> {
+        Some(self.location)
+    }
+}
+
+impl Continuous<f64, f64> for Logistic {
+    /// Calculates the probability density function for the logistic
+    /// distribution at `x`
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// exp(-(x - μ) / σ) / (1 + exp(-(x - μ) / σ))^2 / σ
+    /// ```
+    /// where `μ` is the location and `σ` is the scale
+    fn pdf(&self, x: f64) -> f64 {
+        (-(x - self.location) / self.scale).exp()
+            / (1.0 + (-(x - self.location) / self.scale).exp()).powi(2)
+            / self.scale
+    }
+
+    /// Calculates the log probability density function for the logistic
+    /// distribution at `x`
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// -(x - μ) / σ - 2 * ln(1 + exp(-(x - μ) / σ)) - ln(σ)
+    /// ```
+    ///
+    /// where `μ` is the location and `b` is the scale
+    fn ln_pdf(&self, x: f64) -> f64 {
+        -(x - self.location) / self.scale
+            - self.scale.ln()
+            - 2.0 * (1.0 + (-(x - self.location) / self.scale).exp()).ln()
+    }
```

Comment:
Unless there's an example of a range of values which can more accurately be represented by expanding the logarithm, I prefer not to do that. Multiplications are numerically more stable than additions and are simpler to read. In a range where overflow occurs it might make sense, but dealing with all those cases is unpleasant. If the scale is small, both expressions will run into issues.

Reply:
Updated, let me know if this way will work